packages/@aws-cdk/aws-route53-patterns/lib/vpc-endpoint-service-domain-name.ts

```diff
@@ -0,0 +1,178 @@
+        physicalResourceId: PhysicalResourceId.of(id),
+      },
+      onDelete: {
+        service: 'EC2',
+        action: 'modifyVpcEndpointServiceConfiguration',
+        parameters: {
+          ServiceId: serviceId,
+          RemovePrivateDnsName: true,
+        },
+      },
+      policy: AwsCustomResourcePolicy.fromSdkCalls({
+        resources: AwsCustomResourcePolicy.ANY_RESOURCE,
+      }),
+    });
+
+    // Get the name/value pair for the TxtRecord
+    // We're always going to check the name/value because it's harmless
+    const guaranteeAlwaysLookup = randomString();
+    const getNames = new AwsCustomResource(this, 'GetNames', {
+      onCreate: {
+        service: 'EC2',
+        action: 'describeVpcEndpointServiceConfigurations',
+        parameters: {
+          ServiceIds: [serviceId],
+        },
+        physicalResourceId: PhysicalResourceId.of(guaranteeAlwaysLookup),
+      },
+      onUpdate: {
+        service: 'EC2',
+        action: 'describeVpcEndpointServiceConfigurations',
+        parameters: {
+          ServiceIds: [serviceId],
+        },
+        physicalResourceId: PhysicalResourceId.of(guaranteeAlwaysLookup),
+      },
+      policy: AwsCustomResourcePolicy.fromSdkCalls({
+        resources: AwsCustomResourcePolicy.ANY_RESOURCE,
+      }),
+    });
+    // We only want to call and get the TXT DNS details after we've enabled private DNS
+    getNames.node.addDependency(enable);
+
+    // Here are the name/value we get from the endpoint service
+    const name = getNames.getResponseField('ServiceConfigurations.0.PrivateDnsNameConfiguration.Name');
+    const value = getNames.getResponseField('ServiceConfigurations.0.PrivateDnsNameConfiguration.Value');
+
+    // Create the TXT record in the provided hosted zone
+    const verificationRecord = new TxtRecord(this, 'DnsVerificationRecord', {
+      recordName: name,
+      values: [value],
+      zone: props.publicZone,
+    });
+    // Only try making it once we have the values
+    verificationRecord.node.addDependency(getNames);
+
+    // Tell the endpoint service to verify the domain ownership
+    const startVerification = new AwsCustomResource(this, 'StartVerification', {
+      onCreate: {
+        service: 'EC2',
+        action: 'startVpcEndpointServicePrivateDnsVerification',
+        parameters: {
+          ServiceId: serviceId,
+        },
+        physicalResourceId: PhysicalResourceId.of(Fn.join(':', [name, value])),
+      },
+      onUpdate: {
+        service: 'EC2',
+        action: 'startVpcEndpointServicePrivateDnsVerification',
+        parameters: {
+          ServiceId: serviceId,
+        },
+        physicalResourceId: PhysicalResourceId.of(Fn.join(':', [name, value])),
+      },
+      policy: AwsCustomResourcePolicy.fromSdkCalls({
+        resources: AwsCustomResourcePolicy.ANY_RESOURCE,
+      }),
+    });
+    // Only verify after the record has been created
+    startVerification.node.addDependency(verificationRecord);
```

Comment:
this constructor is getting to be a little unwieldy with all of these custom resources, can you extract them to functions that set up the custom resources?